packages/framework/presence/src/datastoreSupport.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type {
+	FluidObject,
+	IFluidLoadable,
+	IRequest,
+	IResponse,
+} from "@fluidframework/core-interfaces";
+import type { IFluidHandleInternal } from "@fluidframework/core-interfaces/internal";
+import { assert } from "@fluidframework/core-utils/internal";
+import { FluidDataStoreRuntime } from "@fluidframework/datastore/internal";
+import type {
+	IFluidDataStoreContext,
+	IFluidDataStoreFactory,
+} from "@fluidframework/runtime-definitions/internal";
+import { create404Response } from "@fluidframework/runtime-utils/internal";
+
+/**
+ * Extends the FluidDataStoreRuntime to provide a request method that routes requests to
+ * the entrypoint Fluid object that is expected to be a {@link LoadableFluidObject}.
+ */
+class BasicFluidDataStoreRuntime extends FluidDataStoreRuntime {
+	public override async request(request: IRequest): Promise<IResponse> {
+		const response = await super.request(request);
+		if (response.status !== 404) {
+			return response;
+		}
+		// Return entrypoint object if someone requests it directly.
+		// Direct requests exist from two scenarios:
+		//   1. the request url is a "/"
+		//   2. the request url is empty
+		if (request.url === "" || request.url === "/" || request.url.startsWith("/?")) {
```

Comment:
Looks like there's a third case here, 3. request url starts with "/?"

Reply:
I copied this from DataObject and had the same thought. But it really is two cases. Both `/` and `/?` are URL `/`. The `?` and remainder are parameters. I think starts with `?` isn't listed because that is invalid.?